### PR TITLE
[WIP] Improvements for generalization round 2

### PR DESCRIPTION
**Pull request by choldgraf**
A few more improvements to generalize this past just MEG data

### CODE REVIEW

File: mne_bids/utils.py [hunk 1]
```diff
@@ -93,9 +96,13 @@ def make_bids_filename(subject=None, session=None, task=None,
     filename = []
     for key, val in order.items():
         if val is not None:
+            if any(ii in val for ii in ['-', '_']):
+                raise ValueError("Unallowed `-` or `_` found in key/value pair"
+                                 " %s: %s" % (key, val))
```
Comment (jasmainak) [inline mne_bids/utils.py]: a test for this?

**Comment by choldgraf** [comment]
tests are happy!

**Comment by jasmainak** [comment]
The example `create_bids_folder.py` is now broken. I think we should soon enable CircleCI to avoid these kind of issues.

**Comment by choldgraf** [comment]
good call, I got the example working. Why would we need to use Circle to test examples? If we just build the docs as a part of travis then it'll error if one of the scripts errors, no?

**Comment by jasmainak** [comment]
Well CircleCI also gives us the built html so we can look at it if we want. I don't know if it's possible in Travis ... but yeah either would do the job as you say

**Comment by jasmainak** [comment]
Still one more issue:

```py
Traceback (most recent call last):
  File "/home/mainak/anaconda2/lib/python2.7/site-packages/sphinx_gallery/gen_rst.py", line 450, in execute_code_block
    exec(code_block, example_globals)
  File "<string>", line 11, in <module>
  File "/home/mainak/Desktop/projects/github_repos/mne-bids/mne_bids/meg_bids.py", line 319, in raw_to_bids
    suffix='%s%s' % (kind, ext), prefix=data_path)
  File "/home/mainak/Desktop/projects/github_repos/mne-bids/mne_bids/utils.py", line 101, in make_bids_filename
    " %s: %s" % (key, val))
ValueError: Unallowed `-` or `_` found in key/value pair task: visual_faces
```

**Comment by choldgraf** [comment]
updated the example + made a few tweaks, let's see if this passes!

File: mne_bids/utils.py [hunk 2]
```diff
@@ -233,3 +235,11 @@ def _write_json(dictionary, fname, verbose=False):
     if verbose is True:
         print(os.linesep + "Writing '%s'..." % fname + os.linesep)
         print(json_output)
+
+
+def _check_key_val(key, val):
+    """Perform checks on a value to make sure it adheres to the spec."""
+    if any(ii in val for ii in ['-', '_', '/']):
+        raise ValueError("Unallowed `-` or `_` found in key/value pair"
```
Comment (jasmainak) [inline mne_bids/utils.py]: you are missing '/' in the error message

**Comment by choldgraf** [comment]
ok, I believe nitpicks addressed!

**Comment by jasmainak** [comment]
Thanks @choldgraf !

**Pull request merged**